### PR TITLE
[ENG-326] send more build metadata: release channel, build profile, git commit hash

### PR DESCRIPTION
**Pull request by dsokal**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.

# Why

https://linear.app/expo/issue/ENG-326/add-build-profile-to-build-metadata

See also https://github.com/expo/universe/pull/7080

# How

I made EAS CLI send more build metadata to www:
- release channel
- build profile
- git commit hash

# Test Plan

I ran a build and used graphql to fetch the metadata.

### CODE REVIEW

File: packages/eas-cli/src/build/metadata.ts [hunk 1]
```diff
@@ -20,15 +21,18 @@ export async function collectMetadata<T extends Platform>(
...
 ): Promise<BuildMetadata> {
   return {
+    trackingContext: ctx.trackingCtx,
     appVersion: ctx.commandCtx.exp.version!,
     cliVersion: packageJSON.version,
     workflow: ctx.buildProfile.workflow,
     credentialsSource,
     sdkVersion: ctx.commandCtx.exp.sdkVersion,
-    trackingContext: ctx.trackingCtx,
+    releaseChannel: ctx.buildProfile.releaseChannel,
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/metadata.ts:30]: what about a release channel that is not defined in eas.json?
I'm not saying to add that, just to make sure this is intended behaviour

File: packages/eas-cli/src/utils/git.ts [hunk 2]
```diff
@@ -38,7 +43,15 @@ async function doesGitRepoExistAsync(): Promise<boolean> {
   }
 }
 
-async function isGitInstalledAsync(): Promise<boolean> {
+export async function gitCommitHashAsync(): Promise<string | undefined> {
+  try {
+    return await (await spawnAsync('git', ['rev-parse', 'HEAD'])).stdout.trim();
+  } catch (err) {
+    return undefined;
```
Comment (wkozyra95) [inline packages/eas-cli/src/utils/git.ts:50]: When this could happen?

File: packages/eas-cli/src/utils/git.ts [hunk 3]
```diff
@@ -38,7 +43,15 @@ async function doesGitRepoExistAsync(): Promise<boolean> {
   }
 }
 
-async function isGitInstalledAsync(): Promise<boolean> {
+export async function gitCommitHashAsync(): Promise<string | undefined> {
+  try {
+    return await (await spawnAsync('git', ['rev-parse', 'HEAD'])).stdout.trim();
```
Comment (wkozyra95) [inline packages/eas-cli/src/utils/git.ts:48]:
```suggestion
    return (await spawnAsync('git', ['rev-parse', 'HEAD'])).stdout.trim();
```

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/utils/git.ts (hunk 2 above)
Comment (dsokal) [reply packages/eas-cli/src/utils/git.ts:50]: Short answer: I don't know. 
Long answer: Slack

File: packages/eas-cli/src/build/metadata.ts [hunk 4]
```diff
@@ -20,15 +25,46 @@ export async function collectMetadata<T extends Platform>(
...
+  if (ctx.buildProfile.workflow === Workflow.GENERIC) {
+    if (isExpoUpdatesInstalled(ctx.commandCtx.projectDir)) {
+      return 'default';
+    } else {
+      return undefined;
+    }
+  } else {
+    return 'default';
+  }
+}
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/metadata.ts:70]:
```suggestion
async function resolveReleaseChannel<T extends Platform>(
  ctx: BuildContext<T>
): Promise<string | undefined> {
  if (!isExpoUpdatesInstalled(ctx.commandCtx.projectDir)) {
    return undefined;
  }
  if (ctx.buildProfile.releaseChannel) {
    return ctx.buildProfile.releaseChannel;
  }

  let maybeReleaseChannel: string | null;
  if (ctx.platform === Platform.ANDROID) {
    maybeReleaseChannel = await readAndroidReleaseChannelSafelyAsync(ctx.commandCtx.projectDir);
  } else {
    maybeReleaseChannel = await readIosReleaseChannelSafelyAsync(ctx.commandCtx.projectDir);
  }
  if (maybeReleaseChannel) {
    return maybeReleaseChannel;
  }

  return 'default';
}
```

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**